Observer/Checkout/SubmitAllAfter.php

```diff
@@ -97,13 +107,34 @@ public function execute(Observer $observer): void
 
         try {
             $result = $this->subscriptionCreator->createSubscriptions($quote, $order);
-            $this->checkoutSession->setData(SubscriptionCreator::CREATED_SUBSCRIPTION_IDS, $result[SubscriptionCreator::CREATED_SUBSCRIPTION_IDS]);
-            $this->checkoutSession->setData(SubscriptionCreator::FAILED_SUBSCRIPTION_COUNT, $result[SubscriptionCreator::FAILED_SUBSCRIPTION_COUNT]);
+            $this->checkoutSession->setData(
+                SubscriptionCreator::CREATED_SUBSCRIPTION_IDS,
+                $result[SubscriptionCreator::CREATED_SUBSCRIPTION_IDS]
+            );
+            $this->checkoutSession->setData(
+                SubscriptionCreator::FAILED_SUBSCRIPTION_COUNT,
+                $result[SubscriptionCreator::FAILED_SUBSCRIPTION_COUNT]
+            );
         } catch (\Exception $e) {
             $this->logger->critical($e);
             $this->checkoutSession->setData(SubscriptionCreator::CREATED_SUBSCRIPTION_IDS, []);
             $this->checkoutSession->setData(SubscriptionCreator::FAILED_SUBSCRIPTION_COUNT, 0);
         }
+
+        if ($this->isSubscriptionsCreated($result)) {
```

Comment:
`$result` could be undefined if `$this->subscriptionCreator->createSubscriptions($quote, $order)` method throws exception.